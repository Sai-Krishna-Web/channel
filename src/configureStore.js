import {createStore, applyMiddleware} from 'redux';
import createSagaMiddleware from 'redux-saga'
import rootReducer from './reducers';
import rootSaga from './sagas/channelSaga';

const sagaMiddleWare=createSagaMiddleware();
export default ()=>{
    let store=createStore(rootReducer,applyMiddleware(sagaMiddleWare));
    sagaMiddleWare.run(rootSaga);
   return{
        store
    }
}
