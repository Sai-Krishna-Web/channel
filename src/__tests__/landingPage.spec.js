import React from 'react';
import { render } from '@testing-library/react';
import { shallow, mount } from 'enzyme';
import LandingPage from '../components/landingPage';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

const initialState={};
const mockStore = configureStore();
let store1=mockStore(initialState);

let wrapper=shallow(<Provider store={store1}><LandingPage /></Provider>);
let component = mount(<Provider store={store1}><LandingPage /></Provider>);
describe('renders LandingPage Component', () => {
    it('renders correctly', () => {
      expect(wrapper.exists()).toBe(true);
    });

    it('should match the snapshot', () => {
        expect(wrapper).toMatchSnapshot();
      });

    it('should match the product details content', () => {
        
        expect(component.text()).toContain('View Channels');
      });
    
    it('check Button Link exists',()=>{
        expect(component.find('button').exists()).toBe(true);
    });
});

