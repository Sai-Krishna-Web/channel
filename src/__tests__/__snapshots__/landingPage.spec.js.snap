// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders LandingPage Component should match the snapshot 1`] = `ShallowWrapper {}`;
